test/integration/attachment_link_header_test.rb

```diff
@@ -0,0 +1,104 @@
+require 'test_helper'
+require 'capybara/rails'
+
+class AttachmentLinkHeaderIntegrationTest < ActionDispatch::IntegrationTest
+  extend Minitest::Spec::DSL
+  include Capybara::DSL
+  include Rails.application.routes.url_helpers
+  include PublicDocumentRoutesHelper
+
+  let(:filename) { 'sample.docx' }
+  let(:asset_id) { 'asset-id' }
+
+  before do
+    login_as create(:managing_editor)
+    stub_whitehall_asset(filename, id: asset_id, draft: asset_initially_draft)
+  end
+
+  context 'given a file attachment' do
+    let(:file) { File.open(path_to_attachment(filename)) }
+    let(:attachment) { build(:file_attachment, attachable: attachable, file: file) }
+    let(:attachable) { edition }
+
+    before do
+      setup_publishing_api_for(edition)
+      attachable.attachments << attachment
+      VirusScanHelpers.simulate_virus_scan
+    end
+
+    context 'on a draft document' do
+      let(:edition) { create(:news_article) }
+      let(:asset_initially_draft) { true }
+
+      it 'sets link to parent document in Asset Manager when document is published' do
+        visit admin_news_article_path(edition)
+        force_publish_document
+
+        parent_document_url = Whitehall.url_maker.public_document_url(edition)
+
+        Services.asset_manager.expects(:update_asset)
+          .with(asset_id, 'parent_document_url' => parent_document_url)
+
+        AssetManagerUpdateAssetWorker.drain
+      end
```

Comment:
Minor: In [my most recent work on this stuff](https://github.com/alphagov/whitehall/blob/32bb13cf3ae3ae9ca71fa14bd4c620c631660419/test/integration/attachment_replacement_integration_test.rb#L30-L63), I've been moving the "when" bit of the test code into another nested context and then having two "it" tests where firstly the existing behaviour is tested (i.e. in this case requesting the asset and examining the response header) and, secondly, setting the expectation and draining the queue. The idea being to clearly illustrate the behaviour generated by `AttachmentsController#show` and that generated by calls to the Asset Manager API. It _might_ be worth considering doing the same here.

Reply:
I'm not going to do the same at the moment, although I can see/agree that it'd be good to make these tests consistent in future.